lib/spack/spack/test/repo.py

```diff
@@ -667,61 +689,26 @@ def test_repo_descriptors_construct(tmp_path: pathlib.Path):
     repos_2 = spack.repo.RepoDescriptors(descriptors_2)  # type: ignore
     repos_3 = spack.repo.RepoDescriptors(descriptors_3)  # type: ignore
 
-    git_clone_calls = 0
-
-    class MockGit(spack.util.executable.Executable):
```

Comment:
Maybe continue without the dependency on `git` the tool, and instead record the git calls that were made?